### DISCUSSION TITLE
Unable to use EksPodOperator to run pods on an external EKS cluster

### DISCUSSION THREAD

**Discussion by Hen0k** [apache/airflow#31345] [General]
I have airflow running in a K8s cluster with the K8s executor enabled. I am trying to run a Pod on another K8s cluster that is on EKS. I have created 2 Airflow connections using the UI. These are AWS and Kubernetes API credentials. 

This is the dag I am trying with.

``` python
from pendulum import datetime
from airflow import DAG
from airflow.configuration import conf
from airflow.providers.amazon.aws.operators.eks import EksPodOperator

default_args = {
    "owner": "me",
    "depends_on_past": False,
    "start_date": datetime(2023, 5, 1),
    "email_on_failure": False,
    "email_on_retry": False
}


with DAG(
    dag_id="example_remote_kubernetes_pod", schedule="@once", default_args=default_args
) as dag:
    EksPodOperator(
        aws_conn_id=<"my-aws-conn">,
        kubernetes_conn_id=<"my-k8s-conn">,
        cluster_name=<"my-cluster-name">,
        pod_name="airflow-test-pod",
        # namespace=<"my-ns">,
        image="hello-world",
        labels={"<pod-label>": "<label-name>"},
        task_id="demo-one",
        in_cluster=False,
        is_delete_operator_pod=True,
        get_logs=True
    )
```

When I run this a Pod gets created on the cluster Airflow is running on. But it fails to create the other pod on the remote EKS cluster. There is an exception raised `... Invalid connection configuration. Options kube_config_path, kube_config, in_cluster are mutually exclusive. You can only use one option at a time.`

I have tried different permutation of the parameters like:
- removing the in_cluster option. (But it has a default value of False)
- removing the kubernetes_conn_id. (But it tries looking for the default connection id)
- adding kube_config_path and kube_config separately. But they are not valid arguments to this class.

How can I get this to work. Is this exception a bug or am I missing something. It is defined [here](https://github.com/apache/airflow/blob/7ebda3898db2eee72d043a9565a674dea72cd8fa/airflow/providers/cncf/kubernetes/hooks/kubernetes.py#L195).

**Comment by Hen0k** [comment]
@potiuk  got any Ideas on this one?

**Reply by potiuk** [reply]
No.
